src/plugins/unified_search/public/query_string_input/query_string_input.tsx

```diff
@@ -179,12 +179,21 @@ export default class QueryStringInputUI extends PureComponent<QueryStringInputPr
   };
 
   private fetchIndexPatterns = debounce(async () => {
-    const stringPatterns = this.props.indexPatterns.filter(
-      (indexPattern) => typeof indexPattern === 'string'
-    ) as string[];
-    const objectPatterns = this.props.indexPatterns.filter(
-      (indexPattern) => typeof indexPattern !== 'string'
-    ) as DataView[];
+    const { stringPatterns = [], objectPatterns = [] } = groupBy(
+      this.props.indexPatterns || [],
+      (indexPattern) => {
+        if (typeof indexPattern === 'string') {
+          return 'stringPatterns';
+        }
+        if ('type' in indexPattern && 'value' in indexPattern) {
```

Comment:
I am a bit worried that if on some point `DataView` gets a `value` field then this condition would be truthy for `DataView`. Probably could be a hard to find bug

Maybe also worth checking `(!(indexPattern instanceof DataView)`? 

Also suggest we extract this `{ type: 'title' | 'id'; value: string }` as a separate type as it repeats in multiple places and have this condition as a type guard?